### DISCUSSION TITLE
Unable to build ESP8266 for Tasmota version 14.2.0

### DISCUSSION THREAD

**Discussion by Wintespe** [arendst/Tasmota#22068] [Support request]
I have just installed VisualStudioCodex V1.92.2 and PlatformIO V3.3.3 on a fresh Windows10 64 bit computer. Download the Tasmota 14.2.0 Zip.file, unzip it and select in the PlatformIO.  The environment starts to load the settings and build by itself the default version (Tasmota14.2.0). This fails with the following error messages:

```
Resolving tasmota dependencies...
Platform Manager: Installing https://github.com/tasmota/platform-espressif8266/releases/download/2024.06.00/platform-espressif8266.zip
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Platform Manager: espressif8266@2024.6.0 has been installed!
Tool Manager: Installing platformio/toolchain-xtensa @ ~2.40802.0
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tool Manager: toolchain-xtensa@2.40802.200502 has been installed!
Tool Manager: Installing https://github.com/tasmota/Arduino/releases/download/2.7.7/esp8266-2.7.7.zip
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tool Manager: framework-arduinoespressif8266@2.7.7 has been installed!
Tool Manager: Installing https://github.com/tasmota/esptool/releases/download/v4.7.3/esptool.zip
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tool Manager: tool-esptoolpy@4.7.3 has been installed!
Tool Manager: Installing platformio/tool-scons @ ~4.40700.0
Downloading 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Tool Manager: tool-scons@4.40700.0 has been installed!
Updating metadata for the vscode IDE...
UserSideException: Processing tasmota (platform: https://github.com/tasmota/platform-espressif8266/releases/download/2024.06.00/platform-espressif8266.zip; framework: arduino; board: esp8266_1M)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_DOUT
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp8266_1M.html
PLATFORM: Espressif 8266 (2024.6.0) > Espressif Generic ESP8266 Tasmota 1M sketch NO FS
HARDWARE: ESP8266 80MHz, 80KB RAM, 972.00KB Flash
PACKAGES: 
 - framework-arduinoespressif8266 @ 2.7.7 
 - tool-esptoolpy @ 4.7.3 
 - toolchain-xtensa @ 2.40802.200502 (4.8.2)
Converting tasmota.ino
Installing Python dependencies
Collecting wheel>=0.35.1
  Downloading wheel-0.44.0-py3-none-any.whl.metadata (2.3 kB)
Collecting zopfli>=0.2.2
  Downloading zopfli-0.2.3-cp311-cp311-win_amd64.whl.metadata (3.0 kB)
Downloading wheel-0.44.0-py3-none-any.whl (67 kB)
Downloading zopfli-0.2.3-cp311-cp311-win_amd64.whl (97 kB)
Installing collected packages: zopfli, wheel
Successfully installed wheel-0.44.0 zopfli-0.2.3
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 101 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0
|-- Adafruit MAX31865 library @ 1.0.1

> more files

|-- Wire @ 1.0
|-- XPT2046_Touchscreen
Building in release mode
ImportError: DLL load failed while importing zopfli: Das angegebene Modul wurde nicht gefunden.:
  File "C:\Users\Win10\.platformio\penv\Lib\site-packages\platformio\builder\main.py", line 180:
    env.SConscript(env.GetExtraScripts("post"), exports="env")
  File "C:\Users\Win10\.platformio\packages\tool-scons\scons-local-4.7.0\SCons\Script\SConscript.py", line 612:
    return _SConscript(self.fs, *files, **subst_kw)
  File "C:\Users\Win10\.platformio\packages\tool-scons\scons-local-4.7.0\SCons\Script\SConscript.py", line 279:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "C:\Anwendungen\Tasmota-14.2.0\pio-tools\gzip-firmware.py", line 34:
    from zopfli.gzip import compress
  File "C:\Users\Win10\.platformio\penv\Lib\site-packages\zopfli\gzip.py", line 2:
    import zopfli.zopfli
========================= [FAILED] Took 118.23 seconds =========================

Environment    Status    Duration
-------------  --------  ------------
tasmota        FAILED    00:01:58.233
==================== 1 failed, 0 succeeded in 00:01:58.233 ====================


```

I tried to compile **env:tasmota-DE(Tasmota.14.2.0)** and get the same errors.

What is missing to compile Tasmota for ESP8266 in a fresh environment?

**Comment by HofeBY** [comment]
https://github.com/arendst/Tasmota/discussions/21941

I can compile TM14.2.0.3 development under Win10 with VSC Version: 1.92.2 (user setup)
Commit: fee1edb8d6d72a0ddff41e5f71a671c23ed924b9
Date: 2024-08-14T17:29:30.058Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045

see https://github.com/HofeBY/Tasmota-Firmware
read Liesmich-zuerst.txt

HofeBY

**Reply by Wintespe** [reply]
The configuration should be very similar for me, as Windows10, Chrome, etc. are up to date due to a new installation. We have one essential difference: 

**The VisualStudioCode, PlatformIO, the libraries... are also brand new. 
There is no history. Everything is fresh!** 

To test, I then downloaded **Tasmota 11.1.0** and compiled this version **_without_** errors.

**14.2.0** still doesn't work, and just for testing, **version 13.4.0** doesn't work either.

**Reply by HofeBY** [reply]
Sie könnten zum schnellen Test mal eine Firmware von meiner Seite testen.
Ich habe diese siehe Liesmich-zuerst.txt mit TM14.2.0.3 development compiliert.
Entsprechende verwendete configurationsfiles liegen auch auf der Webseite.

Am besten in VSC plattformIO dann C++ deinstallieren, gefolgt von VSC.

löschen von "%homepath%\.platformio" & ".pio"

Rechner Reboot
Neuinstallation VSC & selbst updaten lassen !
Frischen Sourecode nach D:\Tasmota entpacken.
Änderung siehe Liesmich.txt machen
Erster Compilerlauf für alle ESP CPU Varianten noch ohne geänderte configfiles & bei jedem ersten mal ESP CPU Anwahl beim
nachladen nicht unterbrechen !
Nun Configfiles ändern & neu compilieren.

HofeBY

**Reply by Wintespe** [reply]
Thank you very much, I read the article with interest. 

You wrote:
**It is best to uninstall platformIO in VSC then C++, followed by VSC.
Delete "%homepath%.platformio" & ".pio"**

However, this is not a guarantee that all parts have been safely removed.

Now it gets exciting. I have partially implemented the adjustments #21941 and #21964. 

```
[platformio]
; For best Gitpod performance remove the ";" in the next line. Needed Platformio files are cached and installed at first run
core_dir = .platformio
; For unrelated compile errors with Windows it can help to shorten Platformio project path
;workspace_dir = c:\.pio
;extra_configs           = platformio_tasmota_user_env.ini
```

I can compile Tasmota Version 14.2.0 **env:tasmota32C3** now without errors. Except for the strange and inexplicable error, this ESP32 version compile. The ESP8266 version continues to malfunction.

**Reply by HofeBY** [reply]
and "%homepath%.pio" cleared ?
;workspace_dir = c:\.pio changed
to 
workspace_dir = c:\.pio   ?

**Reply by Jason2866** [reply]
You can try to make even shorter `core_dir = .pla` 
Low risk of conflicts since the folder is moved in Tasmota project dir with this setting.

**Reply by Wintespe** [reply]
Here is a short summary for an overview. All components have been freshly installed including the Tasmota versions.


- Windows10 Pro, x64, 22H2
- VisualStudioCode V1.9.2 
- PlatformIO V3.3.3 

    


<html>
<head>

</head>
<body lang="de-DE" link="#000080" vlink="#800000" dir="ltr">
<dl>
	<dl>
		<dd>
		
Tasmota Version | PlatformIO 					Settings | Compile 					Result
-- | -- | --
14.2.0 					env:tasmota32C3 | core_dir 					= .platformio 					workspace_dir 					= c:\.pio | ok
14.2.0 					env:tasmota32C3 | core_dir 					= .platformio | ok
14.2.0 					env:tasmota-DE | No 					change | failed
14.2.0 					env:tasmota-DE | core_dir 					= .platformio 					workspace_dir 					= c:\.pio | failed
14.2.0 					env:tasmota-DE | core_dir 					= .platformio | failed
13.4.0 					 env:tasmota-DE | No 					change | failed
11.1.0 					 env:tasmota-DE | No 					change | ok

</dl>
</body>
</html>


I hope it helps to find the root cause of the problem.

**Reply by Jason2866** [reply]
The problem is known. Windows with its limits in path lengths. Tasmota adds with every version new features. This adds additional code. This code is listed in the compile and link commands.
So the problem will occur more often in newer versions of Tasmota.

**Reply by Wintespe** [reply]
If this is a path problem, then why with ESP8266 and not **ESP32** (see table). The workspace_dir is **not** set to drive `C:\.pio` to shorten the path. How would the problem be caused by additional features in Tasmota? I don't see any long paths in the error messages.

**Reply by Wintespe** [reply]
@ [HofeBY](https://github.com/HofeBY)
Have you ever successfully compiled an ESP8266 version like 14.2.0 env:tasmota-DE or could you try it out? Maybe this gives a hint.

**Reply by Wintespe** [reply]
I took a careful look at the error messages as far as I can.
I think the errors do not come from the compiler These arise of scrpits at the installation of the software modules.
For example, the zopfli module was installed
**Successfully installed wheel-0.44.0 zopfli-0.2.3**

If the module could be installed, access to the path is also given, otherwise it would have failed during installation.

However, methods cannot be called later
**ImportError: DLL load failed while importing zopfli: Das angegebene Modul wurde nicht gefunden.**

This does not look like path problems under Windows but installation problems of the necessary packages. I don't know how that could be solved. I hope the Tasomta team has an idea.

**Reply by HofeBY** [reply]
Ja, gerade deutsches TM14.2.0.3 tasmota.bin & tasmota.bin.gz compiliert siehe LOG.

LOG:
 *  Executing task: C:\Users\Station\.platformio\penv\Scripts\platformio.exe run --environment tasmota 

Processing tasmota (platform: https://github.com/tasmota/platform-espressif8266/releases/download/2024.06.00/platform-espressif8266.zip; framework: arduino; board: esp8266_1M)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_DOUT
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp8266_1M.html
PLATFORM: Espressif 8266 (2024.6.0) > Espressif Generic ESP8266 Tasmota 1M sketch NO FS
HARDWARE: ESP8266 80MHz, 80KB RAM, 972.00KB Flash
PACKAGES:
 - framework-arduinoespressif8266 @ 2.7.7
 - tool-esptoolpy @ 4.7.3
 - toolchain-xtensa @ 2.40802.200502 (4.8.2)
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 101 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0  
|-- Adafruit MAX31865 library @ 1.0.1
|-- Adafruit MCP9808 Library @ 1.1.2 
|-- Adafruit PM25 AQI Sensor @ 1.0.6 
|-- Adafruit seesaw Library @ 1.3.1  
|-- Adafruit SGP30 Sensor @ 1.2.0    
|-- Adafruit SGP40 Sensor @ 1.1.0    
|-- Adafruit VEML7700 Library @ 1.0.0
|-- ESP8266Audio @ 1.9.5
|-- BME8563
|-- C2Programmer @ 1.0.0
|-- headers
|-- cc1101 @ 1.0
|-- ESP8266WebServer @ 1.0
|-- EEPROM 24C128_256_521
|-- ams @ 1.0
|-- I2Cdevlib-Core
|-- LibTeleinfo @ 1.1.7
|-- arduino-mcp2515-1.0.1
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- NTP Library
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- SPL06-007 @ 0.1.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit TSL2591
|-- base64 @ 1.1.1
|-- BME68x
|-- DNSServer @ 1.1.1
|-- DS2480 Library @ 0.1.0
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ESP KNX IP Library @ 0.5.2
|-- ESP8266HTTPClient @ 1.2
|-- ESP8266httpUpdate @ 1.3
|-- ESP8266mDNS @ 1.2
|-- ESP8266SAM @ 1.0.1
|-- ESP8266WiFi @ 1.0
|-- ESPFtpServer @ 0.1.0
|-- Ext-printf @ 1.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- FT5206_Library @ 1.0.0
|-- GT911
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS(esp8266) @ 0.1.0
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- LoRa @ 0.8.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- RadioLib @ 6.4.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- SD(esp8266) @ 2.0.0
|-- ESP8266SdFat @ 1.1.0
|-- Sensirion I2C SEN5X @ 0.2.0
|-- SPI @ 1.0
|-- stm32_flash @ 1.0.0
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 1.0
|-- TM1638plus @ 1.7.0
|-- Joba_Tsl2561 @ 2.0.10
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
|-- WiFiHelper @ 1.0.0
|-- Wire @ 1.0
|-- XPT2046_Touchscreen
Building in release mode
Compiling C:\.pio\build\tasmota\src\tasmota.ino.cpp.o
D:/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_10_scripter.ino:145:17: note: #pragma message: internal special flash script buffer used
 #pragma message "internal special flash script buffer used"
                 ^
Compiling C:\.pio\build\tasmota\lib710\ams\GcmParser.cpp.o
Compiling C:\.pio\build\tasmota\libb05\tls_mini\StackThunk_light.cpp.o
Compiling C:\.pio\build\tasmota\libb05\tls_mini\WiFiClientSecureLightBearSSL.cpp.o
Compiling C:\.pio\build\tasmota\lib8ae\Display_Renderer-gemu-1.0\epdpaint.cpp.o
Compiling C:\.pio\build\tasmota\lib8ae\Display_Renderer-gemu-1.0\renderer.cpp.o
Compiling C:\.pio\build\tasmota\libbcb\Epaper_29-gemu-1.0\epd2in9.cpp.o
Compiling C:\.pio\build\tasmota\lib08b\Epaper_42-gemu-1.0\epd4in2.cpp.o
Compiling C:\.pio\build\tasmota\lib6c6\LedControl\LedMatrix.cpp.o
Compiling C:\.pio\build\tasmota\libe8a\Xlatb_RA8876-gemu-1.0\RA8876.cpp.o
Compiling C:\.pio\build\tasmota\lib0a8\UDisplay\uDisplay.cpp.o
Linking C:\.pio\build\tasmota\firmware.elf
Retrieving maximum program size C:\.pio\build\tasmota\firmware.elf
Checking size C:\.pio\build\tasmota\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [======    ]  58.4% (used 47868 bytes from 81920 bytes)
Flash: [=======   ]  74.5% (used 762392 bytes from 1023984 bytes)
Building C:\.pio\build\tasmota\firmware.bin
Creating BIN file "C:\.pio\build\tasmota\firmware.bin" using "D:\Tasmota\.platformio\packages\framework-arduinoespressif8266\bootloaders\eboot\eboot.elf" and "C:\.pio\build\tasmota\firmware.elf"
bin_map_copy(["C:\.pio\build\tasmota\firmware.bin"], ["C:\.pio\build\tasmota\firmware.elf"])
map_gzip(["C:\.pio\build\tasmota\firmware.bin"], ["C:\.pio\build\tasmota\firmware.elf"])
bin_gzip(["C:\.pio\build\tasmota\firmware.bin"], ["C:\.pio\build\tasmota\firmware.elf"])
Compression reduced firmware size to 70% (was 766544 bytes, now 534418 bytes)
firm_metrics(["C:\.pio\build\tasmota\firmware.bin"], ["C:\.pio\build\tasmota\firmware.elf"])
Used static IRAM: 30832 bytes ( 1936 remain, 94.1 % used)
====================================================================================== [SUCCESS] Took 54.49 seconds ======================================================================================
Environment    Status    Duration
-------------  --------  ------------
tasmota        SUCCESS   00:00:54.488
====================================================================================== 1 succeeded in 00:00:54.488 ====================================================================================== 
 *  Terminal will be reused by tasks, press any key to close it. 

liegt unter https://github.com/HofeBY/Tasmota-Firmware
hab´s noch nicht geflasht, sollte aber OK sein.
DE, SML mit KNX, 8kB SCRIPT, etc.pp aber OHNE RULES !

HofeBY

**Reply by Jason2866** [reply]
Compile run of tasmota-DE with Windows -> https://github.com/Jason2866/Tasmota/actions/runs/10672674597/job/29580548515
Works without any changes in setup.

**Reply by Wintespe** [reply]
Thanks, that looks good. There are no path problems on the part of Windows. In my new environment the paths are of similar length. It turns out that there is an installation problem, by whoever. It could be Tasmota, PLatformIO.  Let's see what happens.

**Reply by Wintespe** [reply]
@ Jason2866
What should I do with **os-check-win (tasmota-DE)** to solve the problem? Were the tests carried out on a freshly installed system?

**Reply by Jason2866** [reply]
Yes, Github Actions provides so called Runners. Always a clean fresh "new" environment where all needed has to be installed. You can compare with a PC where Windows is just is installed.
It is Windows Server 2019. Details are here https://docs.github.com/de/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Reply by Wintespe** [reply]
_You can compare with a PC where Windows is just is installed_.I

It is a virtual machine. Which runner did you use and what was the preinstalled software. However, I see no help to solve the current problem. The problem is real and exists as the errors above show.

**Reply by Jason2866** [reply]
Never doubt you have the problem. I can't help since i can not reproduce a failure.

**Reply by Wintespe** [reply]
I'm familiar with that. However, However, I was hoping that the error messages would give you a clue about the problem and that you might have ideas about how to fix it. I'll look into the ideas and then we'll see what happens. I'm afraid that the problem will crop up somewhere else at some point, for example when switching to Windows 11, which is just around the corner for many poeple.

**Reply by Jason2866** [reply]
Windows 11 is not really different, the functions used and needed by VSC and Platformio are not changed. Stopped using Windows a few years ago. So not really the one which is still in deep with Windows trickery stuff.
Anyways did setup UTM on my Mac M1 and run Windows11 Arm in a VM. After patching Platformio to allow to use the Windows x86_64 tools and toolchains (trough the Windows emulator layer) the env tasmota-DE does (really slow) successfully compile.

**Reply by Wintespe** [reply]
It's not the architecture of Windows 11 but the reinstallation of Visual Studio Code and PlatformIO when changing.
For my part, I had designing Windows applications for industrial control systems and kernel drivers for a long time. So I know Windows pretty well in many of its facets. That doesn't help me at the moment because, as far as I know, VisualStudioCode and PlatformIO have their roots in Linux. Perhaps someone from the community or the Tasmota team has a few ideas. Otherwise, Tasmota developments under Windows could become problematic. We'll see.

**Reply by Jason2866** [reply]
VSC is from Microsoft and has its roots in Windows afaik. Platformio is completely written in Python.
The used tools and toolchains to compile have there roots in Linux. And that's the problem here, since the are not designed to take care how long pathes are. The problem will be solved when Arduino core 3.1.x is used (currently in development). The newer toolchains are build to take care of the Windows path limits. So in future there will be no problems with Windows too.

**Reply by Wintespe** [reply]
VSC is supported by Microsoft, right.
I know. That's what you mean about paths. But in my opinion that's not true because it works for HofeBY,  as he confirmed above. It compiles for me for the 32-bit versions of ESP too. Compiled perfect with deliberately very very long paths in all derivatives.
What do you mean by 
**The newer toolchains are build to take care of the Windows path limits**
 What tools are included? Is there a schedule?

**Reply by Jason2866** [reply]
Arduino core is from espressif and the repo is here https://github.com/espressif/arduino-esp32
Tasmota framework is based on and will be used when the official core 3.1.0 is released.
Actualy Tasmota framework is based on core 3.0.4. and using espressif toolchains gcc esp-12.2.0 from 20230208. Core 3.1.0 will use gcc esp-13.2.0 from 20240530
**BUT** this is only valid for esp32x MCUs. There will be no changes or newer tools and toolchains for the esp8266. The support for the esp8266 from espressif has ended already years ago.

**Reply by Wintespe** [reply]
It's nice to know that there will be further development of the new ESP derivatives. But please keep in mind that older versions of the ESP are also in circulation and are used. Often it is completely sufficient to turn a light on or off. An ESP8266 is definitely sufficient for this. So please do not delete the "old" versions as happened with Tasmota version 14.2.0

**Support of ESP8266 Core versions before 2.7.7 and ESP32 Core versions before 3.0.4 have been removed.**

Thanks, otherwise the old versions can no longer be compiled locally for small adjustments. The cores were surprisingly deleted from the computer without any further notice.

**Reply by Jason2866** [reply]
There is no support for older Tasmota versions. If you want to compile older versions you have to backup the complete needed setup. Once again Tasmota can't control how Platformio or VSC is behaving. Tasmota do not delete anything in the Platformio setup.

**Reply by Wintespe** [reply]
Thanks for the tips and support. I haven't been able to get VSC and PlatformIO to back up together yet. The .platformio was reinstalled, it didn't work or the libs weren't installed. If you don't have any immediate ideas, I think it would be a good idea to open a new discussion thread for this.

**Comment by Jason2866** [comment]
A solution is to install WSL. Compiling is even faster when using WSL.

**Reply by Wintespe** [reply]
Thx, I'm thinking about using Linux Mint in a virtual machine. I've already set it up and now I just need to install the environment. I will report about it.

**Reply by Wintespe** [reply]
Under Linux MInt the installation of the VSC and the PlatformIO ran without any problems. The following compilation of Tasmota version 14.2.0 was error free.
However, I cannot draw any conclusions about Windows or Linux operating systems from this result. The thing is that the implementation of VSC and PlatformIO are different. Therefore, the tests of the Tasmota compilation should be done independently on both platforms, e.g. Windows and Linux. If possible, do a clean reinstall. The "Runner" in the form of Github as a virtual machine is problematic because it may contain unknown elements.

**Reply by Jason2866** [reply]
Be our guest to do this manually job with Windows ;-)  If the CI tests for Windows (it IS a clean install) is not good enough.

**Reply by Wintespe** [reply]
That's what I did, hence this thread. Unfortunately, no changes in any adjustments. I think it's all been discussed.

**Comment by Noschvie** [comment]
> Windows10

@Wintespe, @HofeBY which Windows Edition are you using?

**Reply by Wintespe** [reply]
It is Windows 10 Pro, 64-bit, 22H2

**Reply by HofeBY** [reply]
same at me, automatic Windows updates stoppted with "StopUpdates10"

**Reply by Wintespe** [reply]
Why did you ask? Do you have any idea what to do?

**Reply by HofeBY** [reply]
Noschvie ask & I answered. Maybe he also has also a problem.
At my site compiling is functional.

Did you answered to all my helping´s ?

Maybe you don´t understand the form of a discussion.

Take a look in regedit32
Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem\LongPathsEnabled ->1

**Reply by Wintespe** [reply]
My question implies nothing other than "do you have the same problem".

As my initial question, I have no problem with long file names under Windows. I have tested this very carefully with extremely long path names.

My problem still exists:
**What is missing to compile Tasmota 14.2.0 for ESP8266 in a fresh Windows 10 environment?**

**Comment by Wintespe** [comment]
**Update:**

Tasmota version 14.3.0 has recently been released. I downloaded it and loaded it into VisualStudioCode.

**Result:**

> The Tasmota versions fror ESP8266 now compile without errors in a fresh Windows environment.
> 
> The old cores were no longer deleted from the computer.

Thanks for the update.
